[3.x] Fix non-latin layout scancodes on Linux, adds access to physical scancodes.

bruvzg opened the pull request:
Backport of #18020, without breaking changes.

Existing Functions properties are not renamed, new property is `physical_scancode`.

For the reference, list of changes to existing API (in both cases physical scancode is used only if remapped scancode is `0`):
- `InputEventKey::action_match` uses physical scancode, if remapped scancode is `0` (Physical key are added to `InputMap` with `scancodes == 0`, remapped keys with `physical_scancode == 0`).
- On Linux/X11, physical scancode is used in place remapped, if scancode is `0` (not remapped correctly by `XLookupString`).

Modified test project from 46751: [Test.zip](https://github.com/godotengine/godot/files/6097439/Test.zip)
<details>
  <summary>Screenshots:</summary>
Dvorak:
<img width="852" alt="Screenshot 2021-03-07 at 16 45 10" src="https://user-images.githubusercontent.com/7645683/110246053-a6699700-7f6e-11eb-98e7-9d24f8ec9985.png">
Russian (phonetic) layout:
<img width="852" alt="Screenshot 2021-03-07 at 16 45 32" src="https://user-images.githubusercontent.com/7645683/110246054-a8335a80-7f6e-11eb-9241-610d547b4404.png">
Results are the same on macOS, Linux (Ubuntu 20.10) and Windows.
</details>

Fixes #46751 and fixes #46750

bruvzg committed:
Add `physical_scancode` (keyboard layout independent keycodes) to InputEventKey and InputMap.
Fix non-latin keyboard layout keycodes on Linux/X11 (fallback to physical keycodes).

Faless commented on platform/javascript/os_javascript.cpp:
```diff
@@ -253,6 +253,7 @@ static Ref<InputEventKey> setup_key_event(const EmscriptenKeyboardEvent *emscrip
 	ev->set_echo(emscripten_event->repeat);
 	dom2godot_mod(emscripten_event, ev);
 	ev->set_scancode(dom_code2godot_scancode(emscripten_event->code, emscripten_event->key));
+	ev->set_physical_scancode(dom_code2godot_scancode(emscripten_event->code, emscripten_event->key));
```
There is actually "proper" (quoted because we are still talking about the web :( ) implementation for this. Let me do the backport on top of these changes so you could simply add it on top.

Faless replied:
See https://github.com/Faless/godot/commit/516eacef08ea13686126920a1a27ca484c704da6

bruvzg replied:
Thanks. I have pulled JS commit to the PR branch.

Faless committed:
[HTML5] Add support for `physical_keycode`.

This uses the `event.code` value to retrieve the physical code, while
still using the extra logic to map the unicode value to our keylist,
when computing the `scancode` (supporting ASCII and Latin-1).

akien-mga commented:
Thanks!

jitspoe commented:
Does this mean we'll be able to bind the default keys based on key positions instead of the letters of the keys?  Would be great for non-qwerty keyboard users!

Calinou commented:
> Does this mean we'll be able to bind the default keys based on key positions instead of the letters of the keys? Would be great for non-qwerty keyboard users!

Yes 🙂

jitspoe commented:
Awesome!  I saw there's a physical_key that gets written to the project file, but it seems to always be 0.  Maybe it's only supported in Linux at the moment?

akien-mga commented:
@jitspoe Should work on all platforms, but `key` and `physical_key` are mutually exclusive. You should click "Physical Key" to add one to the input map (and `key` will be 0).
![Screenshot_20210919_112749](https://user-images.githubusercontent.com/4701338/133922411-e81d8ef5-d7bd-4459-8a6a-7e855f734dc7.png)

jitspoe commented:
Is there a way to get the display string from a physical_key?  Can't seem to find anything, so I made a proposal here: https://github.com/godotengine/godot-proposals/issues/3571

agameraaron commented:
I really appreciate this!! Was really hoping I wouldn't have to wait until 4.0 for this and thankfully that is now the case.
